### ISSUE TITLE
Move from vault repository credentials to using pypi trusted publishers

### ISSUE THREAD

**Issue by costrouc** [conda-incubator/conda-store#519]
Required steps that I can think of. @anirrudh please add more to the issue as you run into issues.

 - [ ] switch to using trusted publishing of python packages in `.github/workflows/release.yaml` follow https://blog.pypi.org/posts/2023-04-20-introducing-trusted-publishers/
 - [ ] remove vault logic
 - [ ] speak with @iameskild on adding `conda-incubator/conda-store` as a trusted publisher to the pypi packages `conda-store` and `conda-store-server`

**Comment by anirrudh** [comment]
I think we are already using the trusted publishers mechanisms from the UI side :)

**Comment by costrouc** [comment]
@anirrudh yes but this needs to be enabled for pypi conda-store and conda-store-server which are separate packages on pypi.

**Comment by costrouc** [comment]
I've added conda-incubator/conda-store as a trusted publisher for conda-store and conda-store-server using the release.yaml workflow file.

I've added conda-incubator/conda-store as a trusted publisher for jupyterlab-conda-store using the build_and_deploy_release.yml file.

So now all that should be needed is modifying the workflow

**Comment by trallard** [comment]
Thanks @costrouc - @anirrudh can you please make sure to prioritise this?
